apps/aechannel/src/aesc_fsm_min_depth_watcher.erl

```diff
@@ -978,18 +998,23 @@ tx_location(TxHash, C) ->
     cached_get({location, TxHash}, C, fun(C1) -> tx_location_(TxHash, C1) end).
 
 tx_location_(TxHash, C) ->
-    L = case aec_chain:find_tx_with_location(TxHash) of
+    {L, Type, SignedTx} =
+        case aec_chain:find_tx_with_location(TxHash) of
             none ->
                 lager:debug("couldn't find tx hash", []),
-                undefined;
-            {mempool, _} ->
+                {undefined, undefined, undefined};
```

Comment:
So the third element in this tuple means that `undefined` can be stored as `tx` in the map in `#tx_log_entry.value`? Is this supported flow? If yes, type of field `tx` shall reflect this (I currently see `aetx_sign:signed_tx()`).

Reply:
Actually, it shouldn't be. I was pondering that while making changes before, but didn't change it then.

Reply:
I see addressed at https://github.com/aeternity/aeternity/pull/2488/commits/d04810664b3e91cdba8ee6a94419c9f203e7d812#diff-f609866532cd9838417a435634ee892aR1034